flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalSort.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.flink.table.runtime.operators.sort.ProcTimeSortOperator;
+import org.apache.flink.table.runtime.operators.sort.RowTimeSortOperator;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.TimestampKind;
+import org.apache.flink.table.types.logical.TimestampType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.IntStream;
+
+/** Stream exec RelNode for time-ascending-order Sort without `limit`. */
+public class StreamExecTemporalSort extends ExecNodeBase<RowData>
+        implements StreamExecNode<RowData> {
+
+    private final int[] keys;
+    private final boolean[] orders;
+    private final boolean[] nullsIsLast;
+
+    public StreamExecTemporalSort(
+            int[] keys,
+            boolean[] orders,
+            boolean[] nullIsLast,
+            ExecEdge inputEdges,
+            LogicalType outputType,
+            String description) {
+        super(Collections.singletonList(inputEdges), outputType, description);
+        this.keys = keys;
+        this.orders = orders;
+        this.nullsIsLast = nullIsLast;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected Transformation<RowData> translateToPlanInternal(PlannerBase planner) {
+        TableConfig config = planner.getTableConfig();
+        RowType inputType = (RowType) getInputNodes().get(0).getOutputType();
+        // time ordering needs to be ascending
+        if (orders.length == 0 || !orders[0]) {
+            throw new TableException(
+                    "Sort: Primary sort order of a streaming table must be ascending on time.\n"
+                            + "please re-check sort statement according to the description above");
+        }
+        LogicalType timeType = inputType.getTypeAt(keys[0]);
+        Transformation<RowData> input =
+                (Transformation<RowData>) getInputNodes().get(0).translateToPlan(planner);
+
+        if (timeType instanceof TimestampType) {
+            TimestampType keyType = (TimestampType) timeType;
+            if (keyType.getKind() == TimestampKind.ROWTIME) {
+                return createSortRowTime(inputType, input, config);
+            } else if (keyType.getKind() == TimestampKind.PROCTIME) {
+                return createSortProcTime(inputType, input, config);
+            } else {
+                throw new TableException(
+                        "Sort: Internal Error\n"
+                                + "Normally, this happens unlikely. please contact customer support for this");
+            }
+        } else {
+            throw new TableException(
+                    "Sort: Internal Error\n"
+                            + "Normally, this happens unlikely. please contact customer support for this");
+        }
```

Comment:
merge them into one? and improve the exception message. There is no useful info in the current message.